### PR TITLE
Update Transactions and Webhooks

### PR DESCRIPTION
**Pull request by AlexLakatos**
- added information about Refund Transactions on the Webhooks and Transactions pages
- added sample events for all webhooks except qualifying ones
- changed order of topics on Webhooks page to better match order of working with Webhooks
- added CURL code samples on the Webhooks and Transactions pages
- changed some of the passive voice usages to active voice to improve clarity
- Added extra sections on the Transactions page to break up the content, adding CURL sample as well
- Changed occurrences of "Fidel" to "Fidel API" for SEO purposes
- ran both pages through Grammarly to check my spelling, tone of voice, clarity and delivery. Apparently we're "Very Clear", "Very Engaging" and our delivery is "Just Right" most of the time.

### CODE REVIEW

File: transactions/README.md [hunk 1]
```diff
@@ -97,36 +107,69 @@ fileName:transaction.json
...
+The Fidel API currently supports three types of transactions: authorisation transactions, clearing transactions and refund transactions.
+
+**Authorisation transactions** are processed when a purchase registers on a linked card. For example, when a customer is making a payment in-store or online in real-time. When a customer makes a payment with a linked debit/credit card in an auth-enabled location, the `transaction.auth` webhook is also triggered and the transaction object sent to your specified URL in real-time.
+
+**Clearing transactions**, also known as "settled transactions", are processed when a payment transaction settles, usually happens 48 to 72 hours after a payment registers. The Fidel processes for clearing transactions are also triggering the `transaction.clearing` webhook events. The processes run daily at 12:00 UTC for Mastercard and multiple times per day for Visa and American Express. Only one transaction is sent per event.
```
Comment (c0derabbit) [inline transactions/README.md:114]: can you use smartypants quotes please? 
‘single quotes’
left: option + ], right: option + shift + ]
“double quotes”
left: option + [, right: option + shift + ]

the easier solutions: 
a) i think if you mix these with "regular" quotes in grammarly, gm will ask you to choose one style.
b) some writers, like Bear, automagically makes quotes smart (quite annoyingly in code blocks, too 🤦‍♀️ )

File: transactions/README.md [hunk 2]
```diff
@@ -97,36 +107,69 @@ fileName:transaction.json
...
 </details>
 
-There are two types of transactions: authorisation transactions and clearing transactions. **Authorisation transactions** are processed when the customer is making the payment in-store or online. They trigger the `transaction.auth` webhook event that you can use to notify or reward the user in your application in real-time.
+The Fidel API currently supports three types of transactions: authorisation transactions, clearing transactions and refund transactions.
+
+**Authorisation transactions** are processed when a purchase registers on a linked card. For example, when a customer is making a payment in-store or online in real-time. When a customer makes a payment with a linked debit/credit card in an auth-enabled location, the `transaction.auth` webhook is also triggered and the transaction object sent to your specified URL in real-time.
```
Comment (c0derabbit) [inline transactions/README.md:112]: `is making a payment` -> `makes a payment` flows smoother imo

is the hyphen needed in `in real-time`? it's surely `real-time data` as an adjective, but when something happens `in real time` (or `in store`), i'm not so sure about that. can we double check this with Tom please?

File: transactions/README.md [hunk 3]
```diff
@@ -97,36 +107,69 @@ fileName:transaction.json
...
-We suggest that you use the auth event to notify the user that you registered the transaction and will fulfill the reward when the transaction clears, since the clearing is the confirmation that the transaction has been settled.
+```sh
+curl -X GET \
+  https://api.fidel.uk/v1/transactions/a375e18f-0678-40fa-aa8b-4875e2146437 \
+  -H 'Content-Type: application/json' \
+  -H 'Fidel-Key: sk_test_50ea90b6-2a3b-4a56-814d-1bc592ba4d63'
+```
 
-Please allow up to 24h after linking a Mastercard card to start receiving real-time authorisation transactions.
+We suggest that you use the auth event to notify the user that you registered the transaction and will fulfil the reward when the transaction clears. The clearing event is the confirmation that the transaction has been settled.
```
Comment (c0derabbit) [inline transactions/README.md:131]: 🙏 ❤️

File: transactions/README.md [hunk 4]
```diff
@@ -97,36 +107,69 @@ fileName:transaction.json
...
+## Test Transactions
 
-## Test transactions
+For testing purposes, you can use the [**API Playground**](https://dashboard.fidel.uk/playground) in the Fidel Dashboard test environment to create test transactions and test your application logic. Alternatively, you can use the [Create Test Transaction](https://reference.fidel.uk/reference#create-transaction-test) API endpoint to create authorisation test transaction.
 
-For testing purposes, you can use the **API Playground** in test environment to create test transactions and test your application logic.
+To create a test transaction, you will need a [Program](/programs), a [Location](/locations) and a test [Card](cards) linked to the program.
 
-To create a test transaction you will need a program, a location and a test card linked to the program.  
+##### Create Test Transactions Using the API.
```
Comment (c0derabbit) [inline transactions/README.md:145]: do we need a period after these titles?

File: transactions/README.md [hunk 5]
```diff
@@ -97,36 +107,69 @@ fileName:transaction.json
...
 
-Fidel matches full refund transactions to its original transaction by comparing `cardId`, `locationId`, `merchantId`, `amount` and `datetime`. The refund `transactionId` is set in the original transaction as `refundTransactionId` and the original `transactionId` is set on the refund transaction as `originalTransactionId`.
+## Test Transactions
 
-## Test transactions
+For testing purposes, you can use the [**API Playground**](https://dashboard.fidel.uk/playground) in the Fidel Dashboard test environment to create test transactions and test your application logic. Alternatively, you can use the [Create Test Transaction](https://reference.fidel.uk/reference#create-transaction-test) API endpoint to create authorisation test transaction.
 
-For testing purposes, you can use the **API Playground** in test environment to create test transactions and test your application logic.
+To create a test transaction, you will need a [Program](/programs), a [Location](/locations) and a test [Card](cards) linked to the program.
```
Comment (c0derabbit) [inline transactions/README.md:143]: `[Card](cards)` missing the slash from `/cards`

File: transactions/README.md [hunk 6]
```diff
@@ -97,36 +107,69 @@ fileName:transaction.json
...
+    "cardId": "bc538b71-31c5-4699-820a-6d4a08693314",
+    "locationId": "7a916fbd-70a0-462f-8dbc-bd7dbfbea140"
+  }'
+```
 
-To create a test transaction, use the dropdown menus to select the Program, Location and Card for the transaction.  These selections will be used to populate the `cardId`, `locationId` and the `amount` in the JSON payload.  You can modify any of the properties in the JSON file (including the amount). If the transaction is created successfully you will see the transaction object in the response body box.
+##### Create Test Transactions Using the API Playground.
 
-##### Create sample transactions in test mode using the API Playground.
+On the Fidel Dashboard, go to the [**Playground**](https://dashboard.fidel.uk/playground) option in the navigation menu. Click on **Create transaction** link, located on the left side, in the **END POINTS** menu. The method will be set to POST and the endpoint to **_/transactions/test_**. An editable sample JSON object like the following one will be used to create the transaction.
```
Comment (c0derabbit) [inline transactions/README.md:161]: `END POINTS` -> `ENDPOINTS` please. do we have to capitalise it?

File: transactions/README.md [hunk 7]
```diff
@@ -97,36 +107,69 @@ fileName:transaction.json
...
+
+##### Clear Test Transactions Using the Dashboard.
+
+To clear an authorisation test transaction, you can navigate to the [Transaction](https://dashboard.fidel.uk/transactions) option in the Dashboard navigation menu. You'll see all your transactions listed there. Find the one you want to change the status of, click on the three dots on the right side of it. A popup menu will appear, and you should click on the 'Clear transaction' option. This action will change the status of the transaction from `auth` to `cleared`, and trigger the `transaction.clearing` webhook event.
+
+## API Reference
 
-In order to clear an authorisation test transaction you can open the transaction options menu in the dashboard and click on 'Clear transaction'. This action will trigger the `transaction.clearing` webhook event.
+If you're looking to find out more about our Transactions API and how to use it with your application, please visit the [Fidel API Reference](https://reference.fidel.uk/reference#create-transaction-test).
```
Comment (c0derabbit) [inline transactions/README.md:175]: have you thought about `If you're looking to find out more` -> `To find out more`? or is that too dry?

File: webhooks/README.md [hunk 8]
```diff
@@ -1,69 +1,70 @@
 # Webhooks
 
-Fidel uses webhooks to notify your application when relevant events happen in your account. This functionality can be used to receive data from events not triggered by direct API requests or to receive the data in a service that is not responsible for making the API request but needs to consume the response.
+The Fidel API uses [webhooks](https://en.wikipedia.org/wiki/Webhook) to notify your application when relevant events happen in your account. You can use this functionality to receive data from events not triggered by direct API requests. You can also use this functionality to receive the data in a service that is not responsible for making the API request but needs to consume the response.
```
Comment (c0derabbit) [inline webhooks/README.md:3]: too much word repetition in `You can (also)? use this functionality`

File: webhooks/README.md [hunk 9]
```diff
@@ -1,69 +1,70 @@
...
 
-Fidel will notify your registered webhook URLs as the event happens. For example, when a customer makes a payment with a linked Mastercard on a participating location, a `transaction.auth` event is sent in real-time to the specified webhook URL with the transaction object in the request payload.
+There are currently nine webhooks available to use with the Fidel API, each corresponding to an event happening in your account: `brand.consent`, `card.linked `, `card.failed `, `program.status`, `transaction.auth`, `transaction.auth.qualified`, `transaction.clearing`, `transaction.clearing.qualified` and `transaction.refund`. Each of them requires an HTTPS URL to be registered.
```
Comment (c0derabbit) [inline webhooks/README.md:5]: I would avoid counting how many there are, this is something we will forget to update at some point

File: webhooks/README.md [hunk 10]
```diff
@@ -1,69 +1,70 @@
...
 
-## Signatures
-To confirm that received events are being sent from Fidel we recommend verifying webhook signatures. That can be done by using the `x-fidel-signature` and `x-fidel-timestamp` HTTP headers. This isn't required, but offers an additional layer of security.
+There are two ways you can register Webhooks with the Fidel API. You can create them in the [Fidel Dashboard, under the "Webhooks" page](https://dashboard.fidel.uk/webhooks) or make HTTP requests using the [Webhooks API](https://reference.fidel.uk/v1/reference#create-webhook-brand).
 
-A unique secret key is generated for each webhook. The key is returned in the response's `secretKey` property if you are using the Webhooks API. You can also copy the key from the dashboard's webhooks page by clicking in the **Show Key** button next to your webhook endpoint. To verify a webhook request, generate a signature using the same key that Fidel uses and compare that to the value of the `x-fidel-signature` header:
+The Fidel API has two types of webhooks, both of which work similarly, but require slightly different things to be registered.
```
Comment (c0derabbit) [inline webhooks/README.md:24]: `slightly different things` sounds a bit sloppy, can we specify `things`?

File: webhooks/README.md [hunk 11]
```diff
@@ -72,70 +73,40 @@ curl -X POST \
...
-  "Warning",
-  "Fidel-Account",
-  "Fidel-Key",
-  "Fidel-Live",
-  "Fidel-Request-Id",
-  "Fidel-User",
-  "X-Fidel-Signature",
-  "X-Fidel-Timestamp"
-]
+["Accept", "Accept-Charset", "Accept-Encoding", "Accept-Language", "Accept-Datetime", "Access-Control-Request-Method", "Access-Control-Request-Headers", "Cache-Control", "Connection", "Content-Length", "Content-Type", "Cookie", "Date", "Expect", "Forwarded", "From", "Host", "If-Match", "If-Modified-Since", "If-None-Match", "If-Range", "If-Unmodified-Since", "Max-Forwards", "Origin", "Pragma", "Proxy-Authorization", "Range", "Referer", "Te", "Transfer-Encoding", "User-Agent", "Upgrade", "Via", "Warning", "Fidel-Account", "Fidel-Key", "Fidel-Live", "Fidel-Request-Id", "Fidel-User", "X-Fidel-Signature", "X-Fidel-Timestamp"]
```
Comment (c0derabbit) [inline webhooks/README.md:101]: I preferred this with newlines, this is not so readable inline

Comment (AlexLakatos) [reply webhooks/README.md:101]: My concern here is space. It takes too much space with new lines.

File: transactions/README.md (hunk 6 above)
Comment (AlexLakatos) [reply transactions/README.md:161]: The Dashboard says "END POINTS", capitalized and with a space, so I'd like to keep it the same in the docs.

Comment (c0derabbit) [reply transactions/README.md:161]: uh oh, then it's wrong on the Dashboard, it will be fixed in the next version

**Comment by c0derabbit** [comment]
lgtm 👍

**Comment by AlexLakatos** [comment]
Thanks! Dev Previews aren't set up yet, we'll talk about it this week.

**Pull request merged**